code/datums/status_effects/buffs.dm

```diff
@@ -265,67 +265,96 @@
-			new /obj/item/rod_of_asclepius(owner.loc)
-			qdel(owner)
-	else
-		if(ishuman(owner))
-			var/mob/living/carbon/human/itemUser = owner
-			//Because a servant of medicines stops at nothing to help others, lets keep them on their toes and give them an additional boost.
-			if(itemUser.health < itemUser.maxHealth)
-				new /obj/effect/temp_visual/heal(get_turf(itemUser), COLOR_HEALING_GREEN)
-			itemUser.adjustBruteLoss(-1.5)
-			itemUser.adjustFireLoss(-1.5)
-			itemUser.adjustToxLoss(-1.5)
-			itemUser.adjustOxyLoss(-1.5)
-			itemUser.adjustStaminaLoss(-1.5)
-			itemUser.adjustBrainLoss(-1.5)
-			itemUser.adjustCloneLoss(-0.5) //Becasue apparently clone damage is the bastion of all health
-		if(heal_points < max_heal_points)
-			heal_points = min(heal_points += 3, max_heal_points)
-		//Heal all those around you, unbiased
-		for(var/mob/living/L in view(7, owner))
-			if(heal_points <= 0)
-				break
-			if(L.health < L.maxHealth)
-				new /obj/effect/temp_visual/heal(get_turf(L), COLOR_HEALING_GREEN)
-			if(iscarbon(L))
-				L.adjustBruteLoss(-3.5)
-				L.adjustFireLoss(-3.5)
-				L.adjustToxLoss(-3.5)
-				L.adjustOxyLoss(-3.5)
-				L.adjustStaminaLoss(-3.5)
-				L.adjustBrainLoss(-3.5)
-				L.adjustCloneLoss(-1) //Becasue apparently clone damage is the bastion of all health
+			deathTick++
+			return
+
+		owner.visible_message("<span class='notice'>[owner]'s soul is absorbed into the rod, relieving the previous snake of its duty.</span>")
+		var/mob/living/simple_animal/hostile/retaliate/poison/snake/healSnake = new(owner.loc)
+		var/list/chems = list("bicaridine", "perfluorodecalin", "kelotane")
+		healSnake.poison_type = pick(chems)
+		healSnake.name = "Asclepius's Snake"
+		healSnake.real_name = "Asclepius's Snake"
+		healSnake.desc = "A mystical snake previously trapped upon the Rod of Asclepius, now freed of its burden. Unlike the average snake, its bites contain chemicals with minor healing properties."
+		new /obj/effect/decal/cleanable/ash(owner.loc)
+		new /obj/item/rod_of_asclepius(owner.loc)
+		qdel(owner)
+		return
+
+	// A servant of medicines stops at nothing to help others, let's give them an additional boost
+	if(ishuman(owner))
+		var/mob/living/carbon/human/H = owner
+
+		if(H.getBruteLoss() || H.getFireLoss() || H.getOxyLoss() || H.getToxLoss() || H.getBrainLoss() || H.getStaminaLoss() || H.getCloneLoss())
+			H.adjustBruteLoss(-1.5)
+			H.adjustFireLoss(-1.5)
+			H.adjustOxyLoss(-1.5)
+			H.adjustToxLoss(-1.5)
+			H.adjustBrainLoss(-1.5)
+			H.adjustStaminaLoss(-1.5)
+			H.adjustCloneLoss(-0.5) // Because apparently clone damage is the bastion of all health
+			new /obj/effect/temp_visual/heal(get_turf(H), COLOR_HEALING_GREEN)
+
+	// Regenerate points passively
+	if(heal_points < max_heal_points)
+		heal_points = min(heal_points += 3, max_heal_points)
+
+	// The main course: heal everyone around you!
+	for(var/mob/living/L in view(7, owner))
+		if(heal_points <= 0)
+			break
+
```

Comment:
```suggestion
```
Whoops I forgot this, this is just personal preference, so it really is up to you